blockchain/integrity/integrity_test.go

```diff
@@ -1393,12 +1434,61 @@ func TestLoadBlockchainfromDB(t *testing.T) {
 	cfg.Chain.ProducerPrivKey = "308193020100301306072a8648ce3d020106082a811ccf5501822d0479307702010104202d57ec7da578b98dad465997748ed02af0c69092ad809598073e5a2356c20492a00a06082a811ccf5501822da14403420004223356f0c6f40822ade24d47b0cd10e9285402cbc8a5028a8eec9efba44b8dfe1a7e8bc44953e557b32ec17039fb8018a58d48c8ffa54933fac8030c9a169bf6"
 	cfg.Chain.EnableAsyncIndexWrite = false
 	cfg.Genesis.AleutianBlockHeight = 3
+	cfg.Genesis.ToBeEnabledBlockHeight = 5
 
 	t.Run("load blockchain from DB", func(t *testing.T) {
 		testValidateBlockchain(cfg, t)
 	})
 }
 
+// verify the block contains all tx/log indices up to txIndex and logIndex
+func verifyTxLogIndex(dao blockdao.BlockDAO, blk *block.Block, txIndex int, logIndex uint32) bool {
```

Comment:
`*require.Assertions` would be passed into the subfunc, otherwise the location is unknown if the code in subfunc fails

Reply:
updated